runner/components/io.siddhi.langserver/io.siddhi.langserver.launcher/src/main/java/StdioLauncher.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.siddhi.langserver.SiddhiLanguageServer;
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.launch.LSPLauncher;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+/**
+ * Launcher for language server.
+ */
+public class StdioLauncher {
+
+    /**
+     * Main method is executed at the classpath to start Siddhi language server.
+     *
+     * @param args language server/client options
+     * @throws InterruptedException
+     * @throws ExecutionException
+     */
+    public static void main(String args[]) throws InterruptedException, ExecutionException {
```

Comment:
do u want this main class ?

Reply:
> do u want this main class ?

Yes. This is the general structure that has been followed by other language server implementations as well. A sperate launcher class is used for launching the language server-client connection.  For instance,  ballerina launcher implementation is as follows
https://github.com/ballerina-platform/ballerina-lang/blob/a1fd9f45de540387ca6ad687f7d8c2b625b830e1/language-server/modules/launchers/stdio-launcher/src/main/java/org/ballerinalang/langserver/launchers/stdio/Main.java#L32-L48